snakemake/script.py

```diff
@@ -109,25 +109,22 @@ def log_fmt_shell(self, stdout=True, stderr=True, append=False):
         return lookup[(stdout, stderr, append)].format(self.log)
 
 
-class REncoder:
-    """Encoding Pyton data structures into R."""
-
-    @classmethod
-    def encode_numeric(cls, value):
-        if value is None:
-            return "as.numeric(NA)"
-        return str(value)
+class BaseEncoder:
```

Comment:
I realized that there is a lot of redundant code for the PythonEncoder (which I added) the REncoder and the JuiaEncoder, so instead I made a BaseEncoder class that has the common functions.